pin/pin.go

```diff
@@ -519,7 +520,9 @@ func (p *pinner) PinWithMode(c *cid.Cid, mode PinMode) {
 	}
 }
 
-func hasChild(ds mdag.LinkService, root *cid.Cid, child *cid.Cid) (bool, error) {
+// hasChild recursively looks for a Cid among the children of a root Cid.
+// The visit function can be used to shortcut already-visited branches.
+func hasChild(ds mdag.LinkService, root *cid.Cid, child *cid.Cid, visit func(*cid.Cid) bool) (bool, error) {
```

Comment:
Yeah, in general I don't mind having the function declaration be long. Simply breaking at the most convenient place onto the next line is fine. Breaking it out into separate lines per parameter brings back unpleasant memories of really really old C code.